src/CodeGenerator/AutoloadGenerator.php

```diff
@@ -7,87 +7,94 @@
 
     /**
-     * @param string $namespace
+     * @var string
      */
+    private $namespace;
+
     public function __construct(string $namespace)
     {
         $this->namespace = $namespace;
     }
 
-    private function generateAutoloaderClass(array &$classmap)
+    private function writeFile(string $filename, string $code) : void
     {
-        $class = new ClassGenerator('Autoloader');
-        $classmapValue = new PropertyValueGenerator(
-            $classmap,
-            PropertyValueGenerator::TYPE_ARRAY_SHORT,
-            PropertyValueGenerator::OUTPUT_MULTIPLE_LINE
-        );
+        try {
+            $file = new SplFileObject($filename, 'w');
+            $file->fwrite($code);
+        } catch (Throwable $e) {
+            throw new GenerateCodeException(sprintf('Failed to write output file "%s"', $filename), 0, $e);
+        }
+    }
 
-        $registerCode = 'if (!$this->registered) {'.PHP_EOL
-            . '    spl_autoload_register($this);'.PHP_EOL
-            . '    $this->registered = true;'.PHP_EOL
-            . '}'.PHP_EOL
-            . 'return $this;';
+    private function buildFromTemplate(string $templateFile, string $outputFile, array $replacements) : void
+    {
+        $template = file_get_contents($templateFile);
+        $code = strtr($template, $replacements);
+        $outputFile = $this->outputDirectory . '/' . $outputFile;
 
-        $unregisterCode = 'if ($this->registered) {'.PHP_EOL
-            . '    spl_autoload_unregister($this);'.PHP_EOL
-            . '    $this->registered = false;'.PHP_EOL
-            . '}'.PHP_EOL
-            . 'return $this;';
+        $this->writeFile($outputFile, $code);
+    }
 
-        $loadCode = 'if (isset($this->classmap[$class])) {'.PHP_EOL
-            . '    include __DIR__ . \'/\' . $this->classmap[$class];'.PHP_EOL
-            . '}';
+    private function generateClassmapCode(array &$classmap) : string
+    {
+        $lines = array_map(
+            function (string $class, string $file): string {
+                return sprintf(
+                    '%s => %s,',
+                    var_export($class, true),
+                    var_export($file, true)
+                );
+            },
+            array_keys($classmap),
+            $classmap
+        );
 
-        $class
-            ->addProperty('registered', false, PropertyGenerator::FLAG_PRIVATE)
-            ->addProperty('classmap', $classmapValue, PropertyGenerator::FLAG_PRIVATE)
-            ->addMethod('register', [], MethodGenerator::FLAG_PUBLIC, $registerCode)
-            ->addMethod('unregister', [], MethodGenerator::FLAG_PUBLIC, $unregisterCode)
-            ->addMethod('load', ['class'], MethodGenerator::FLAG_PUBLIC, $loadCode)
-            ->addMethod('__invoke', ['class'], MethodGenerator::FLAG_PUBLIC, '$this->load($class);');
+        $indent = str_repeat(' ', 8);
+        return implode("\n$indent", $lines);
```

Comment:
I'd rather avoid string interpolation anywhere in the codebase.

Also, the method call is replaceable with a constant here

Reply:
Indeed.

Reply:
One way to resolve this:

```php
$indent = sprintf("\n%s", str_repeat(' ', 8));
return implode($indent, $lines);
```